### DISCUSSION TITLE
Installation error - Windows and Visual Studio

### DISCUSSION THREAD

**Discussion by julienguegan** [openvinotoolkit/openvino_notebooks#265] [Q&A]
Hi all,

I am trying to `pip install -r requirements.txt` and I have an error when building the library **editdistance**, it says : 

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30037\include\cstdlib(12): fatal error C1083: Cannot open include file: 'math.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Professional\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for editdistance
```

Does someone know how to resolve ? It seems to come from file missing in Visual Studio, I have try to add some components but I can't really figure how to find a solution ...

Thank you for your help !

**Comment by raymondlo84** [comment]
Are you using Python 3.7? I remember seeing the same error before.

**Reply by julienguegan** [reply]
No, I use python 3.8.5

**Reply by ryanloney** [reply]
Hi @julienguegan unfortunately the editdistance dependency requires C++ when using Python 3.8. It does not require C++ when you use Python 3.6 or 3.7. If you have not tried this already, I suggest you install the Visual C++ package from the guide here: https://github.com/openvinotoolkit/openvino_notebooks/wiki/Windows#3-install-c-for-python-38

When I installed it on my Windows laptop, it required a reboot before I could pip install the requirements. 

I have also been able to use Python 3.7 and avoid C++ requirement on Windows. 

Hope this helps!

**Reply by julienguegan** [reply]
Yes, I have Visual C++ installed but from my understanding it feels that it missing some files in my installation ... I guess I can try to re-install it or do it on another PC

**Comment by helena-intel** [comment]
@julienguegan Do you have the full Visual Studio? I am not sure if the full version of Visual Studio C++ adds all the required libraries to the path. The redistributable from the page linked above (here is a direct link: https://aka.ms/vs/16/release/vc_redist.x64.exe ) does do that (after rebooting your system). Could you try that?

**Reply by julienguegan** [reply]
Actually, I am not really sure that I want to re-install Visual Studio because I have already a lot of code C++/python that is running with my current version and I will prefer to find a solution without losing my configuration (as I have the 2019 version and specific component installed)

**Comment by helena-intel** [comment]
@julienguegan Understood, I did not mean re-installing Visual Studio though. There is a difference between Visual Studio, the development environment that includes C++ compilers, and the VS redistributable, that only includes the libraries that you need to run applications that were built with it.

As ryanloney mentioned above: the easiest way to work around this is to install Python 3.7 on Windows. That does not require the C++ redistributable. This installer should work alongside Python 3.8, so you can keep Python 3.8 in your path and for your other Python development. To use Python 3.7 you can use `py -3.7`. For example: `py -3.7 -m venv openvino_env`.  (It would be best to delete the openvino_env folder created with Python 3.8 first) https://www.python.org/ftp/python/3.7.9/python-3.7.9-amd64.exe

**Reply by julienguegan** [reply]
I get it, Thank you a lot for your support ! I will try what you said §
